src/Model/Post.php

```diff
@@ -263,7 +263,10 @@ public static function selectOriginal(array $fields = [], array $condition = [],
 			return $result;
 		}
 
-		return self::selectFirst($original_fields, ['uri-id' => $result['thr-parent-id'], 'uid' => [0, $result['uid']]], $params);
+		$final_query_condition = ['uri-id' => $result['thr-parent-id']];
+		$final_query_condition = DBA::mergeConditions($final_query_condition, ['uid != 0']);
```

Comment:
This looks dangerous to me.

Reply:
> This looks dangerous to me.

We need the original post from the thread. The 0 uid definitely didn't make it. We can't assume the user making the request did either. My theory was that the earliest post not by uid 0 would be the one that originally shared it. If we select the post by 0 user then I think that will cause problems further down when we try to add interactions to it. I arrived at this idea just looking at the raw table for a couple of posts so maybe there is a logic error in there and/or there could be an easier way to do this. The original way isn't working though so we have to change it.

Reply:
What's the status of this discussion @annando ?

Reply:
I have to have a deeper look at it, but currently I had got some time issues.

Reply:
I had a look at it. This change isn't needed. We mostly don't query any user related data. There seems to be only y single query - and that one is ordered by the `uid', so we fetch the one with a filled `uid` first.

Reply:
The big part of the change is not having the 0 user and making sure it is the oldest one. When the post id that came back was the one for the 0 user it was messing things up. This was happening on a very sparsely populated isolated test Friendica so maybe it wouldn't happen in production? I'll see if I can reproduce to confirm that without the change I got a post from the 0 user.

Reply:
I can't seem to recreate the problem I was seeing earlier so I am reverting this to the original state.